spec/controllers/pockets_controller_spec.rb

```diff
@@ -186,4 +185,71 @@ def edit_weight_call(pocket_id, weight, token)
       end
     end
   end
+
+  describe 'PUT #add_weight' do
+    let!(:unweighed_pocket) do
+      create(:unweighed_pocket, organization: organization, collection: collection)
+    end
+    let!(:weighed_pocket) do
+      create(:weighed_pocket, organization: organization, collection: collection)
+    end
+    let!(:device) do
+      create(:device, device_id: '1', device_type: 'android', organization: organization)
+    end
+
+    def add_weight_call(pocket_id, weight, token)
+      @request.headers['ApiKey'] = token
```

Comment:
mmm you should use the helper which is already created for creating authenticated users.